config/routes.rb

```diff
@@ -9,6 +9,11 @@
   get "/users/:id", to: "users#show", as: "user"
   
   get "/users/:id/posts", to: "posts#index", as: "user_posts"
-  get "/users/:id/posts/:id", to: "posts#show", as: "user_post"
-
+  get "/posts/new", to: "posts#new", as: "user_posts_new"
+  post "/posts", to: "posts#create", as: "user_posts_create"
+  get "/posts/:id/new", to: "comments#new", as: "user_post_comments_new"
+  post "/posts/:id", to: "comments#create", as: "user_post_comments_create"
+  post "/posts/:id/like", to: "likes#create", as: "user_post_likes"
+  get "/posts/:id", to: "posts#show", as: "user_post"
```

Comment:
- [ ] [optional] Great work implementing the changes requested, however, I would recommend the use of resources in your routes, this will make your short easier to read, and scaling up in the future will be easier.
 eg
```suggestion
   resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new, :create] do
      resources :comments, only: [:new, :create]
      resources :likes, only: [:create, :destroy]
    end
  end
  root 'users#index'
```